apollo-router/src/configuration/mod.rs

```diff
@@ -129,7 +136,36 @@ impl Configuration {
             apollo_plugins: ApolloPlugins {
                 plugins: apollo_plugins,
             },
+            dev,
+        }
+    }
+
+    pub(crate) fn enable_dev_mode(&mut self) {
+        if std::env::var("APOLLO_ROVER").ok().as_deref() == Some("true") {
+            tracing::info!("Development mode has been enabled. This mode of operation is only meant for development!");
+        } else {
+            tracing::warn!("Development mode has been enabled and has not been started by `rover dev`. This mode of operation is only meant for development!");
+        }
+
+        if self.plugins.plugins.is_none() {
+            self.plugins.plugins = Some(Map::new());
         }
+
+        self.plugins.plugins.as_mut().unwrap().insert(
+            "experimental.expose_query_plan".to_string(),
+            Value::Bool(true),
+        );
+        self.plugins.plugins.as_mut().unwrap().insert(
+            "experimental.include_subgraph_errors".to_string(),
+            json!({"all": true}),
+        );
```

Comment:
This `insert` will silently overwrite any existing configuration for these plugins. Do we want to warn users if their configuration is being overwritten/ignored? It seems like it might be a good thing.

Also: a document comment on the function to note that this must run after normal configuration processing for plugins has finished.

Reply:
actually it's a good catch, if it's specified in the config file it shouldn't be overriden. fixed